Geometry/VeryForwardGeometryBuilder/interface/DetGeomDesc.h

```diff
@@ -30,62 +37,111 @@ class CTPPSRPAlignmentCorrectionData;
+  const RotationMatrix& rotation() const { return m_rot; }
+
+  // shape info
+  // params() is left for general access to solid shape parameters, but should be used
+  // only with great care, for two reasons: 1. order of parameters may possibly change from
+  // a version to another of DD4hep; 2. length parameters unit is cm while PPS uses mm.
+  const std::vector<double>& params() const { return m_params; }  // default unit: mm from oldDD, cm from DD4hep
+  bool isABox() const { return m_isABox; }
+  const DiamondDimensions& getDiamondDimensions() const {
+    if (!isABox()) {
+      edm::LogError("DetGeomDesc::getDiamondDimensions is not called on a box, for solid ")
+          << name() << ", Id = " << geographicalID();
+    }
+    return m_diamondBoxParams;
+  }  // in mm
+
+  // sensor type
+  const std::string& sensorType() const { return m_sensorType; }
 
-  /// components (children) management
+  // ID info
+  DetId geographicalID() const { return m_geographicalID; }
+
+  // components (children) management
+  const Container& components() const { return m_container; }
+  float parentZPosition() const { return m_z; }  // in mm
   void addComponent(DetGeomDesc*);
   bool isLeaf() const { return m_container.empty(); }
 
-  /// geometry information
-  RotationMatrix rotation() const { return m_rot; }
-  Translation translation() const { return m_trans; }
-  const std::string& name() const { return m_name; }
-  std::vector<double> params() const { return m_params; }
-  int copyno() const { return m_copy; }
-  const std::string& sensorType() const { return m_sensorType; }
-
-  /// alignment
+  // alignment
   void applyAlignment(const CTPPSRPAlignmentCorrectionData&);
 
+  void print() const;
+
 private:
-  DetGeomDesc() {}
-  void deleteComponents();      /// deletes just the first daughters
-  void deepDeleteComponents();  /// traverses the treee and deletes all nodes.
+  void deleteComponents();      // deletes just the first daughters
+  void deepDeleteComponents();  // traverses the tree and deletes all nodes.
   void clearComponents() { m_container.resize(0); }
 
-  Container m_container;
-  Translation m_trans;
-  RotationMatrix m_rot;
-  std::string m_name;
-  std::vector<double> m_params;
-  DetId m_geographicalID;
+  std::string computeNameWithNoNamespace(const std::string_view nameFromView) const;
+  std::vector<double> computeParameters(const cms::DDFilteredView& fv) const;
+  DiamondDimensions computeDiamondDimensions(const bool isABox,
+                                             const bool isDD4hep,
+                                             const std::vector<double>& params) const;
+  DetId computeDetID(const std::string& name, const std::vector<int>& copyNos, unsigned int copyNum) const;
+  DetId computeDetIDFromDD4hep(const std::string& name, const std::vector<int>& copyNos, unsigned int copyNum) const;
+  std::string computeSensorType(const std::string_view name);
```

Comment:
Just to note that `std::string_view` provides `const`-only interface to the pointed-to string, so qualifying it as `const` is unnecessary. (technically it doesn't harm either, so I suppose the main effect is raising the question "why is it `const`")

Reply:
Ha yes thanks, will remove the const. No reason, just historically it was a const std::string&, and then changed quickly to std::string_view because needed by sth else.